src/util.py

```diff
@@ -260,12 +265,160 @@ def xr_add_cyclic_longitudes(da, coord):
+            This is the instaance of CRS that the coordinates will be transformed to.
+        clabel_locations (:class:`list`):
+            List of coordinate tuples in GPS form (lon in degrees, lat in degrees)
+            that specify where the contours with regular field variable values should be plotted.
+        fontsize (:class:`int`):
+            Font size of contour labels.
+        whitebbox (:class:`bool`):
+            Setting this to "True" will cause all labels to be plotted with white backgrounds
+        horizontal (:class:`bool`):
+            Setting this to "True" will cause the contour labels to be horizontal.
+    Returns:
+        cLabels (:class:`list`):
+            List of text instances of all contour labels
+    """
+    import numpy as np
+    import matplotlib.pyplot as plt
+
+    # Initialize empty array that will be filled with contour label text objects and returned
+    cLabels = []
+
+    # Plot any regular contour levels
+    if clabel_locations != []:
+        clevelpoints = proj.transform_points(transform,
+                                             np.array([x[0] for x in clabel_locations]),
+                                             np.array([x[1] for x in clabel_locations]))
+        transformed_locations = [(x[0], x[1]) for x in clevelpoints]
+        ax.clabel(contours, manual=transformed_locations, inline=True, fontsize=fontsize, colors='k', fmt="%.0f")
+        [cLabels.append(txt) for txt in contours.labelTexts]
+
+        if horizontal is True:
+            [txt.set_rotation('horizontal') for txt in contours.labelTexts]
+
+    if whitebbox is True:
+        [txt.set_bbox(dict(facecolor='w', edgecolor='none', pad=2)) for txt in cLabels]
+
+    return cLabels
+
+
+def plotELabels(da, contours, transform, ax, proj, clabel_locations=[], eType='Low', fontsize=22, horizontal=True, whitebbox=False):
+
+    """
+    Utility function to plot contour labels. High/Low contour labels will be plotted using text boxes for more accurate label values
+    and placement.
+
+    Args:
+        da: (:class:`xarray.DataArray`):
+            Xarray data array containing the lat, lon, and field variable data values.
+        contours (:class:`cartopy.mpl.contour.GeoContourSet`):
+            Contour set that is being labeled.
+        transform (:class:`cartopy._crs`):
+            Instance of CRS that represents the source coordinate system of coordinates.
+            (ex. ccrs.Geodetic()).
+        ax (:class:`matplotlib.pyplot.axis`):
+            Axis containing the contour set.
+        proj (:class:`cartopy.crs`):
+            Projection 'ax' is defined by.
+            This is the instaance of CRS that the coordinates will be transformed to.
+        clabel_locations (:class:`list`):
+            List of coordinate tuples in GPS form (lon in degrees, lat in degrees)
+            that specify where the contour labels should be plotted.
+        type (:class:`list`):
```

Comment:
Rather than having a parameter 'type' that indicates whether you are labeling highs or lows, would it make more sense to simply have a text parameter 'label', defaulting to "L", that would allow the user to specify whatever label they like. E.g. "L", "Low", or something else entirely?

Reply:
Ok, I changed it so the user can choose the text part of the label.

Reply:
Great. One more change: "pressure" -> "data"

Then should be good to go!